tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -3,50 +3,151 @@ const { When, Then } = require('cucumber')
     const settings = settingsTable.rowsHash()
-    return client.page.FilesPageElement
+    await client.page.FilesPageElement
       .filesList()
       .closeSidebar(100)
       .openPublicLinkDialog(resource)
-      .addNewLink(settings.role)
+    return client.page.FilesPageElement.publicLinksDialog().addNewLink(settings)
   }
 )
 
 When('the public uses the webUI to access the last public link created by user {string}', async function (linkCreator) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
   return client.page.publicLinkFilesPage().navigateAndWaitTillLoaded(lastShareToken)
 })
 
-When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+When('last public share of the user {string} is deleted using API', async function (sharer) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  const lastShareId = lastShare.id
+  return sharingHelper.deleteLastPublicShare(sharer, lastShareId)
+})
 
-  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
+Then('user {string} should not have a public link with last-share-id', function (sharer) {
+  return sharingHelper.assertUserHasNotShareWithLastShareId(sharer, lastShareID)
+})
+
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
 
+When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
+  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
   return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
 })
 
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('the public link should not have been generated', function () {
+  return client.page.FilesPageElement
+    .publicLinksDialog()
+    .assertPublicLinkNotCreated()
+})
+
+Then('the public should not get access to the publicly shared file', function () {
+  return client
+    .page
+    .publicLinkPasswordPage()
+    .assertResourceAccessDenied()
+})
+
+When('the user edits the public link named {string} of file/folder/resource {string} changing following without save',
```

Comment:
What is the difference between with save and without save? Both do the same thing.